st-tech/fluent-bit-pubsub-custom
# Change Pub/Sub Authentication from JWT to ADC, Improve Build Process, and Remove Unnecessary Files

## Summary
- Change the authentication method for Google Cloud Pub/Sub from JWT to ADC Search.
- This change makes it easier to choose the best authentication method based on the situation, including user accout and metadata server authentication.
- Revised the build process to be independent of the host architecture, as the previous method depended on the host's architecture.
- Removed unnecessary build-generated files from the repository. 

## Changes
- Removed authentication code using JWT and updated it to use ADC.
- Updated related test code to verify the functionality of ADC authentication.
- Revised the build process to be independent of the host architecture.
- Removed build-generated files such as .so and .h files.

## Verification Steps
1. Set the environment variable `GOOGLE_APPLICATION_CREDENTIALS`.
2. Run tests and ensure all tests pass.
3. Deploy the application and verify that communication with Pub/Sub works correctly.

## Additional Information
- For more details on ADC authentication, refer to the official Google Cloud documentation: [Google Cloud Application Default Credentials](https://cloud.google.com/docs/authentication/application-default-credentials)
